specs/_features/focil/beacon-chain.md

```diff
@@ -0,0 +1,277 @@
+
+*Note:* This specification is built upon [Electra](../../electra/beacon_chain.md) and is under active development.
+
+## Constants
+
+### Domain types
+
+| Name | Value |
+| - | - |
+| `DOMAIN_IL_COMMITTEE`       | `DomainType('0x0C000000')`  # (New in FOCIL)|
+
+## Preset
+
+### Inclusion List Committee
+
+| Name | Value | 
+| - | - | 
+| `IL_COMMITTEE_SIZE` | `uint64(2**9)` (=512)  # (New in FOCIL) |
+
+### Execution
+
+| Name | Value |
+| - | - |
+| `MAX_TRANSACTIONS_PER_INCLUSION_LIST` |  `uint64(1)` # (New in FOCIL) TODO: Placeholder | 
+
+## Containers
+
+### New containers
+
+#### `InclusionSummary`
+
+```python
+class InclusionSummary(Container):
+    address: ExecutionAddress
+    nonce: uint64
+    gas_limit: uint64
+```
+
+#### `LocalInclusionList`
+
+```python
+class LocalInclusionList(Container):
+    slot: Slot
+    validator_index: ValidatorIndex
+    parent_hash: Hash32
+    summaries: List[InclusionSummary, MAX_TRANSACTIONS_PER_INCLUSION_LIST]
+    transactions: List[Transaction, MAX_TRANSACTIONS_PER_INCLUSION_LIST]
+```
+
+#### `SignedLocalInclusionList`
+
+```python
+class SignedLocalInclusionList(Container):
+    message: LocalInclusionList
+    signature: BLSSignature
+```
+
+#### `InclusionSummaryAggregate`
+
+```python
+class InclusionSummaryAggregate(Container):
+    aggregation_bits: Bitvector[IL_COMMITTEE_SIZE]
+    summary: InclusionSummary
+    signature: BLSSignature
+```
+
+#### `IndexedInclusionListAggregate`
+
+```python
+class IndexedInclusionListAggregate(Container):
+    validator_indices: List[ValidatorIndex, IL_COMMITTEE_SIZE]
+    summary: InclusionSummary
+    signature: BLSSignature
+```
+
+#### `InclusionSummaryAggregates`
+
+```python
+class InclusionSummaryAggregates(Container):
+    List[InclusionSummaryAggregate, MAX_TRANSACTIONS_PER_INCLUSION_LIST]
+```
+
+### Modified containers
+
+#### `BeaconBlockBody`
+
+**Note:** The Beacon Block body is modified to contain a new `inclusion_summary_aggregate` field.
+
+```python
+class BeaconBlockBody(Container):
+    randao_reveal: BLSSignature
+    eth1_data: Eth1Data  # Eth1 data vote
+    graffiti: Bytes32  # Arbitrary data
+    # Operations
+    proposer_slashings: List[ProposerSlashing, MAX_PROPOSER_SLASHINGS]
+    attester_slashings: List[AttesterSlashing, MAX_ATTESTER_SLASHINGS_ELECTRA]
+    attestations: List[Attestation, MAX_ATTESTATIONS_ELECTRA]
+    deposits: List[Deposit, MAX_DEPOSITS]
+    voluntary_exits: List[SignedVoluntaryExit, MAX_VOLUNTARY_EXITS]
+    sync_aggregate: SyncAggregate
+    bls_to_execution_changes: List[SignedBLSToExecutionChange, MAX_BLS_TO_EXECUTION_CHANGES]
+    # FOCIL
+    inclusion_summary_aggregates: InclusionSummaryAggregates   # [New in FOCIL]
+```
+
+### Beacon State accessors
```

Comment:
I don't think that's part of the state transition function though. I briefly mentioned this in the validator spec. I think this is an implementation detail. Nodes will cache signed local inclusion lists, aggregate them, and release them

Reply:
I don't think it belongs to beacon state transition function either. I think it belongs to beacon state accessors or at least a helper. Currently, all beacon state accessors are using `InclusionSummaryAggregate` so I think they need some function to derive it from the received a list of `SignedLocalInclusionList`. It's for attesters rather than IL committee members.

Reply:
I don't think it belongs to beacon state transition function either. I think it belongs to beacon state accessors or at least a helper. Currently, all beacon state accessors are using InclusionSummaryAggregate so I think they need some function to derive it from the received a list of SignedLocalInclusionList. It's for attesters rather than IL committee members.

Reply:
Let's add that to the fork choice spec or validator spec. Fork choice spec will use it to lock views